### PR TITLE
Expose all input parameters to template as JSON

### PR DESCRIPTION
**Pull request by mark9white**
This change allows you to use {{input.parameters}} to fetch the full parameters list as a JSON string.

### CODE REVIEW

**Comment by xianlubird** [comment]
Maybe input.parameters.<NAME> is not enough ?

**Comment by mark9white** [comment]
I'm not sure what you mean?

**Comment by xianlubird** [comment]
Since now we can get parameter from `inputs.parameters.<NAME>`, so I don't know your case about getting the whole parameters

**Comment by mark9white** [comment]
If you have the following parameters:
- name: name1
  value: value1
- name: name2
  value: value2

In the case where you want to pass them all into the pod rather than specific parameters, then:

{{inputs.parameters}}

will resolve to the entire parameter set as JSON:

[{name: "name1", value: "value1"}, {name: "name2", value: "value2"}]

This is especially useful combined with https://github.com/argoproj/argo/pull/1490

File: workflow/common/util.go [hunk 1]
```diff
@@ -325,6 +325,12 @@ func substituteParams(tmpl *wfv1.Template, globalParams, localParams map[string]
 		}
 		replaceMap["inputs.parameters."+inParam.Name] = *inParam.Value
 	}
+	//allow {{inputs.parameters}} to fetch the entire input parameters list as JSON
+	jsonInputParametersBytes, err := json.Marshal(globalReplacedTmpl.Inputs.Parameters)
+	if err != nil {
+		return nil, errors.InternalWrapError(err)
+	}
+	replaceMap["inputs.parameters"] = string(jsonInputParametersBytes)
```
Comment (xianlubird) [inline workflow/common/util.go]: Should we limit the string length,  I'm afraid the whole parameters will be too long for env

**Comment by mark9white** [comment]
Interesting point … I believe maximum command length in Linux is 2Mb, so I
don’t think this would normally be an issue.

Are you suggesting a straight truncation? Or leaving it empty if it’s too
long?


On 19 July 2019 at 10:11:57, Xianlu Bird (notifications@github.com) wrote:

*@xianlubird* commented on this pull request.
------------------------------

In workflow/common/util.go
<https://github.com/argoproj/argo/pull/1488#discussion_r305269009>:

> @@ -325,6 +325,12 @@ func substituteParams(tmpl *wfv1.Template, globalParams, localParams map[string]
                }
                replaceMap["inputs.parameters."+inParam.Name] = *inParam.Value
        }
+       //allow {{inputs.parameters}} to fetch the entire input
parameters list as JSON
+       jsonInputParametersBytes, err :=
json.Marshal(globalReplacedTmpl.Inputs.Parameters)
+       if err != nil {
+               return nil, errors.InternalWrapError(err)
+       }
+       replaceMap["inputs.parameters"] = string(jsonInputParametersBytes)

Should we limit the string length, I'm afraid the whole parameters will be
too long for env

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/argoproj/argo/pull/1488?email_source=notifications&email_token=AABO7LRQ4SIQ3E5ARZPZFULQAGANZA5CNFSM4IE4D3LKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB665GXA#pullrequestreview-264098652>,
or mute the thread

**Comment by xianlubird** [comment]
@jessesuen WDYT ?

File: workflow/common/util.go (hunk 1 above)
Comment (Ark-kun) [reply workflow/common/util.go]: Usually all parameters end up in the command-line anyways (usually prefixed by parameter flags). Passing them as a single argument won't significantly. change the command-line length.

**Pull request merged**